### PR TITLE
[JS] Implements getResourceInformation (was: implements getAllImageUrls)

### PR DESCRIPTION
**Pull request by dclaux**

### CODE REVIEW

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 1]
```diff
@@ -563,6 +563,10 @@ export abstract class CardElement {
         return [];
     }
 
+    getAllImageUrls(): Array<string> {
```
[the remark this answers is not in the document]
Comment (andrewleader) [reply source/nodejs/adaptivecards/src/card-elements.ts]: Clearly we did a poor job updating the spec as information changed, that's something we're learning from and working on improving across the board. The spec seemed to be marked approved even though the spec was only in a "proposed" solution state, and then changes were made during implementation that weren't reflected in the spec (nor shared across the broader team).

That's our bad, as I said previously, apologies that spec wasn't up-to-date before you coded this up.

[the remark this answers is not in the document]
Comment (dclaux) [reply source/nodejs/adaptivecards/src/card-elements.ts]: Yes, sorry I missed that you had mentioned the spec issue in your comment.

I can make the changes, but I'd like to understand why that particular design has been chosen. How is the type property useful? What good is the mimeType property if 99% of the time it's going to be empty?

**Comment by dclaux** [comment]
For the sake of getting this done in a timely manner, I have implemented the same model as .NET. However:
- I believe the "type" property (not "mimeType") is useless, and I would like to remove it
- And I actually didn't implement the "type" property but instead an "owner" property which is a reference to the CardElement or Action that owns the resource

**Comment by dclaux** [comment]
Adding a couple more things to this PR:
- Parse Image.backgroundColor
- Don't drop actions that don't have a title (fixes https://github.com/Microsoft/AdaptiveCards/issues/1893)

**Pull request merged**